spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepositoryFactory.java

```diff
@@ -15,9 +15,19 @@
  */
 package org.springframework.cloud.config.server.environment;
 
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.SSLContext;
 import javax.servlet.http.HttpServletRequest;
 
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
```

Comment:
Is Apache http client a required dependency?

Reply:
I believe so. The Spring HttpComponentsClientHttpRequestFactory depends on it and this is where I'm hooking in to configure SSL validation on the RestTemplate.

Reply:
While it is the recommended way, there is no required apache httpclient dependency. It happens to be on the classpath for jgit
```
[INFO] +- org.eclipse.jgit:org.eclipse.jgit:jar:4.8.0.201706111038-r:compile
[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.5:compile
```

https://github.com/spring-projects/spring-framework/blob/master/spring-web/spring-web.gradle#L47

You'll need to move this to auto-configuration with a `@ConditionalOnClass(HttpClient)` (spring-cloud-netflix has to do this) with a setter perhaps on `VaultEnvironmentRepository` to set the `RestTemplate`.